engine/dlib/src/test/test_httpclient.cpp

```diff
@@ -848,6 +848,53 @@ TEST_P(dmHttpClientTest, MaxAgeCache)
     ASSERT_EQ(dmHttpCache::RESULT_OK, cache_r);
 }
 
+TEST_P(dmHttpClientTest, TrustCache)
+{
+    dmHttpClient::Delete(m_Client);
+
+    // Reinit client with http-cache
+    dmHttpClient::NewParams params;
+    params.m_Userdata = this;
+    params.m_HttpContent = dmHttpClientTest::HttpContent;
+    params.m_HttpHeader = dmHttpClientTest::HttpHeader;
+    params.m_DNSChannel = m_DNSChannel;
+    params.m_TrustCache = true;
+    dmHttpCache::NewParams cache_params;
+    cache_params.m_Path = "tmp/cache";
+    dmHttpCache::Result cache_r = dmHttpCache::Open(&cache_params, &params.m_HttpCache);
+    ASSERT_EQ(dmHttpCache::RESULT_OK, cache_r);
+    m_Client = dmHttpClient::New(&params, m_URI.m_Hostname, m_URI.m_Port, strcmp(m_URI.m_Scheme, "https") == 0);
+    ASSERT_NE((void*) 0, m_Client);
+
+    dmHttpClient::Result r;
+    r = dmHttpClient::Get(m_Client, "/max-age-cached");
+    ASSERT_TRUE((r == dmHttpClient::RESULT_OK && m_StatusCode == 200) || (r == dmHttpClient::RESULT_NOT_200_OK && m_StatusCode == 304));
```

Comment:
Given the max age, it shouldn't really be a 304 right?

Reply:
Why not?

Reply:
You are right (I think). The max age should control whether the http request should be sent.
And then it might still return a 304.

Reply:
Yes, I think so too. From MDN:

"HTTP 304 Not Modified client redirection response code indicates that there is no need to retransmit the requested resources. It is an implicit redirection to a cached resource."